pkg/gadgets/socket-collector/tracer/tracer.go

```diff
@@ -0,0 +1,268 @@
+// Copyright 2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package tracer
+
+import (
+	"bufio"
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/cilium/ebpf/link"
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+	"github.com/vishvananda/netns"
+	"golang.org/x/sys/unix"
+)
+
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang IterTCPv4 ./bpf/tcp4-collector.c -- -I../../.. -O2 -g -c -x c
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang IterUDPv4 ./bpf/udp4-collector.c -- -I../../.. -O2 -g -c -x c
+
+const label = "Gadget socket-collector"
+
+// From https://github.com/torvalds/linux/blob/v5.13/include/net/tcp_states.h#L12
+const (
+	TCP_ESTABLISHED = iota + 1
+	TCP_SYN_SENT
+	TCP_SYN_RECV
+	TCP_FIN_WAIT1
+	TCP_FIN_WAIT2
+	TCP_TIME_WAIT
+	TCP_CLOSE
+	TCP_CLOSE_WAIT
+	TCP_LAST_ACK
+	TCP_LISTEN
+	TCP_CLOSING
+	TCP_NEW_SYN_RECV
+)
+
+type Socket struct {
+	Protocol            string
+	LocalAddress        string
+	LocalPort           uint16
+	RemoteAddress       string
+	RemotePort          uint16
+	Status              string
+	KubernetesNamespace string
+	KubernetesPodname   string
+}
+
+func increaseRlimit() error {
+	limit := &unix.Rlimit{
+		Cur: unix.RLIM_INFINITY,
+		Max: unix.RLIM_INFINITY,
+	}
+	return unix.Setrlimit(unix.RLIMIT_MEMLOCK, limit)
+}
+
+func prifixedLog(s string) {
```

Comment:
I guess this was intended to be prefixedLog rather than prifixed?

Reply:
Function was removed and now errors are passed back to `operationError` so that user is informed about what happened.